packages/telemetry/src/slog-to-otel.js

```diff
@@ -501,60 +623,155 @@ export const makeSlogToOtelKit = (tracer, overrideAttrs = {}) => {
-              const cause = { name, context: syscall.spanContext() };
+              const cause = { name: details, context: syscall.spanContext() };
               kernelPromiseToSendingCause.set(result, JSON.stringify(cause));
             }
             break;
           }
           case 'invoke': {
-            const [_, target, method] = ksc;
-            makeSyscallSpan(`D(${target}).${method}`, { target, method });
+            const [_, target, method, args] = ksc;
+            const sattrs = {
+              'call.target': target,
+              'call.method': method,
+              'call.slots': args.slots.join(','),
+              details: `${detailPrefix}: D(${target}).${method}`,
+            };
+            makeSyscallSpan(name, sattrs);
+            break;
+          }
+          case 'resolve': {
+            const [_, vatID, resolutions] = ksc;
+            const resolves = [];
+            const rejects = [];
+            for (const r of resolutions) {
+              const [kpid, rejected] = r;
+              if (rejected) {
+                rejects.push(kpid);
+              } else {
+                resolves.push(kpid);
+              }
+            }
+            const detailses = [];
+            if (resolves.length) {
+              detailses.push(`resolve(${resolves.join(',')})`);
+            }
+            if (rejects.length) {
+              detailses.push(`reject(${rejects.join(',')})`);
+            }
+            const sattrs = {
+              vatID,
+              details: `${detailPrefix}: ${detailses.join(', ')}`,
+            };
+            makeSyscallSpan(name, sattrs);
+            break;
+          }
+          case 'subscribe': {
+            const [_, vatID, kpid] = ksc;
+            const sattrs = {
+              vatID,
+              kpid,
+              details: `${detailPrefix}: subscribe(${kpid})`,
+            };
+            makeSyscallSpan(name, sattrs);
+            break;
+          }
+          case 'vatstoreGet': {
+            const [_, vatID, key] = ksc;
+            const sattrs = {
+              vatID,
+              key,
+              details: `${detailPrefix}: vatstoreGet('${key}')`,
+            };
+            makeSyscallSpan(name, sattrs);
+            break;
+          }
+          case 'vatstoreGetAfter': {
+            const [_, vatID, priorKey, lowerBound, upperBound = ''] = ksc;
+            const sattrs = {
+              vatID,
+              priorKey,
+              lowerBound,
+              upperBound,
+              details: `${detailPrefix}: vatstoreGetAfter('${priorKey}', '${lowerBound}', '${upperBound}')`,
+            };
+            makeSyscallSpan(name, sattrs);
+            break;
+          }
+          case 'vatstoreSet': {
+            const [_, vatID, key, value] = ksc;
+            const sattrs = {
+              vatID,
+              key,
+              value,
+              details: `${detailPrefix}: vatstoreSet('${key}', '${value}')`,
```

Comment:
Nit: `value` is stringified already, it'd make sense to not quote it:

```suggestion
              details: `${detailPrefix}: vatstoreSet('${key}', ${value})`,
```

Reply:
No. The value of `value` is always a string, so you need the quotes in the output to see where the string starts and ends.  In particular, it's especially hard to read otherwise if the value contains spaces or is an empty string.

Reply:
Oh I though the value was always `JSON.stringify`ed, in which case the quotes would have been redundant.

Reply:
No, the value is simply a string -- in some cases it's produced by a result from `JSON.stringify` (e.g., when the value is something that's been serialized) but that's by no means always the case.